capture/crawler_estadao.py

```diff
@@ -0,0 +1,247 @@
+				u"setembro": u"Sep", u"outubro": u"Oct", u"novembro": u"Nov", u"dezembro": u"Dec"
+			 }
+
+	try:
+		time = soup.findAll("p", {"class":"data"})[0].text
+	except:
+		try:
+			time = soup.findAll("span", {"class":"data"})[0].text
+		except:
+			logger.error('wrong data tags')
+			return None
+	
+	try:
+		time = time.strip().split()
+		time[1] = time[1].lower()
+		time[1] = MONTHS[time[1]]
+		
+		if url.find("estadao.com.br/noticias/") is not -1:
+			time = time[0:3] + time[4:6]
+			time[3] = time[3][0:2]
+			
+		elif url.find("estadao.com.br/blogs/") is not -1:
+			time[3] = time[4][0:2]
+			time[4] = time[4][3:5]
+	except:
+		logger.error('wrong data extraction')
+		return None
+	
+	time = '-'.join(time)
+	
+	try:
+		published_time = datetime.datetime.strptime(time, '%d-%b-%Y-%H-%M')
+		return published_time
+	except:
+		logger.error('wrong published time format')
+		return
+
+def extract_title(soup):
+	""" Extract the news title.
+
+	"""
+	
+	try:
+		title = soup.findAll('h1', {'class':'titulo'})[0].text
+	except:
+		try:
+			title = soup.findAll('h2', {'class':'subtitulo'})[0].text
+		except:
+			logger.error('wrong title tag or attribute')
+			return
+
+	return title
+
+def extract_content(url, response_content):
+	""" Extract relevant information about news page
+
+	:param url: news page's url
+	:param response_content:
+	:type url: string
+	:type response_content:
+	:return: compressed content, title, published time and body content of news page 
+	:rtype: dict()
+
+	"""
+	
+	cleaned_content = clean_content(response_content)
+	
+	if url.find("estadao.com.br/noticias/") is not -1:
+		soupy = Soupy(cleaned_content).find("div", {"itemprop":"articleBody"})
+		try:
+			content = soupy.children.each(Q.text.strip()).filter(len).val()
+		except:
+			try:
+				soupy = Soupy(cleaned_content).find("article")
+				content = soupy.children.each(Q.text.strip()).filter(len).val()
+			except:
+				logger.error("wrong content tags or attributes")
+				return
+
+	elif url.find("estadao.com.br/blogs/") is not -1:
+		try:
+			soupy = Soupy(cleaned_content).find("article")
+			content = soupy.children.each(Q.text.strip()).filter(len).val()
+		except:
+			logger.error("wrong tags or attributes")
+			return
+
+	article = " ".join(content) 
+	return article
+	
+def download_article(url):
+	""" Download the html content of a news page
+
+	:param url: news page's url
+	:type url: string
+	:return: news page's content 
+	:rtype: requests.models.Response
+
+	"""
+
+	article = {
+		'link': url,
+		'source': 'crawler_estadao'
+	}
+
+	logger.info("Downloading article: %s", url)
+	try:
+		response = requests.get(url, timeout=30)
+	except ConnectionError:
+		logger.error("Failed to fetch: %s", url)
+		return
+	except Timeout:
+		logger.error("Timed out while fetching %s", url)
+		return
+	
+	encoding = response.encoding if response.encoding is not None else 'utf8'
+	response_content = response.content.decode(encoding)
```

Comment:
I'm pretty sure requests does that by itself if you use `reponse.text` instead of `response.content`. I know that's true for python3, but not sure if it is for python2. It's probably worth it to check it out.